src/state/notifications/sagas.ts

```diff
@@ -22,4 +42,31 @@ export function* createNotificationEmailSaga(action: CreateNotificationEmailActi
   }
 }
 
-export default [takeEvery(NotificationAction.CreateNotificationEmail, createNotificationEmailSaga)];
+export function* setNotificationEmailSaga(action: SetNotificationEmailAction | unknown) {
+  const { meta, payload } = action as SetNotificationEmailAction;
+  const email = payload.email;
+  try {
+    yield StoreService.setStoreValue('email', email);
+    yield put(setNotificationEmailSuccess(email));
+    const verifyCode = yield call(() => verifyEmail({ email }));
```

Comment:
shouldn't we store email after successfully verifying it?

Reply:
as logic from @Kamill90 I let keep this but we can change after finish if u feel is not necessary

Reply:
I don't get it, you are the author of PR. No one will remember to change it later. My question is why do we save email before verifying it?

Reply:
sure now catch your question. Yes I move after success verify

Reply:
what about action `yield put(setNotificationEmailSuccess(email));`? I guess it should be also moved after saving the email in the store

Reply:
yes I can But I suspect some problem on transition between 2 screens. Some delay from api can coresponding for navigation. but we can try and wait for test

Reply:
so what will happen if the verification will fail? won't be there a bug because the `yield put(setNotificationEmailSuccess(email));` will send incorrect information that everything went ok?

Reply:
good point we dont have now clear way to inform user about error during fail on email veryfication - some simple "Your email can't verify", I can prepare this in simple way and use setNotificationEmailError but suspect to make some more work after marge work onboarding and settings

Reply:
would be nice to add it to this PR

Reply:
is done